internal/configs/node/network.go

```diff
@@ -63,7 +63,13 @@ const (
 	DefaultMaxConnPerIP = 10
 	// DefaultMaxPeers is the maximum number of remote peers, with 0 representing no limit
 	DefaultMaxPeers = 0
-	// DefaultWaitForEachPeerToConnect sets the sync configs to connect to neighbor peers one by one and waits for each peer to connect
+	// DefaultConnManagerLowWatermark is the lowest number of connections that'll be maintained in connection manager
+	DefaultConnManagerLowWatermark = 160
+	// DefaultConnManagerHighWatermark is the highest number of connections that'll be maintained in connection manager
+	// When the peer count exceeds the 'high watermark', as many peers will be pruned (and
+	// their connections terminated) until 'low watermark' peers remain.
+	DefaultConnManagerHighWatermark = 192
```

Comment:
so in here, it kind of conflict with the `DefaultMaxPeers = 0` above

Reply:
> I am curious here, is that connManager redundant with :
> 
> ```
> p2p.MaxConnsPerIP
> p2p.MaxPeers
> ```

these two options are for security manager which is able to control the number of connections per each IP. So, any IP can't have more than a certain number of connections to node. 
ibp2p peer ID is based only on RSA public key. This allows an attacker to create unlimited number of nodes on a single machine (same IP address) and use these nodes to monopolize the incoming/outgoing connections of victim nodes. That's why we use security manager.